astropy/units/tests/test_quantity_ufuncs.py

```diff
@@ -1242,6 +1243,27 @@ def test_two_argument_ufunc_outer(self):
         assert np.all(s13_greater_outer == check13_greater_outer)
 
 
+@dataclasses.dataclass
```

Comment:
Would it make sense to make a `DuckQuantity2(DuckQuantity)` that defines `__array_ufunc__` to check that it in fact does get the chance to operate on it?